.travis.yml

```
@@ -1,20 +1,36 @@
-dist: bionic
+dist: focal
 
-language: node_js
-
-node_js:
-  - "12"
+language: minimal
 
 stages:
   - lint
+  - test
   - deploy
 
+if: branch = master OR type = pull_request
+
+before_install:
+  - tmpdaemon=$(mktemp)
+  - sudo jq '."registry-mirrors" += ["https://mirror.gcr.io"]' /etc/docker/daemon.json > "$tmpdaemon"
+  - sudo mv "$tmpdaemon" /etc/docker/daemon.json
+  - sudo systemctl daemon-reload
+  - sudo systemctl restart docker
+  - docker system info
+
 jobs:
   include:
     - stage: lint
       name: Project Syntax Verification
-      script: make && make docker/run target=lint
-    - stage: deploy
+      script: make docker/run target=lint
+    - stage: test
+      name: Apply Terraform test configs in mockstack
+      install:
+        - make docker-compose/install
+        - make mockstack/up
+      script: make mockstack/pytest
+      after_script:
+        - make mockstack/clean
+   - stage: deploy
```

Oh geez, here it is. Bad yaml...
```suggestion
    - stage: deploy
```

---

I found it by going to the "requests" page in travis-ci, and seeing the "Could not parse" message... https://travis-ci.com/github/plus3it/terraform-aws-tardigrade-config/requests

---

Good eye!  Maybe we should add that to the lint steps at some future date.

---

I was thinking something similar, but I'm not sure how to have the CI validate a parsing error in the CI's own config... It has to parse the config to run the test in the first place... But we could at least provide a target that we could run _locally_ to validate yaml though. Or even use the [`travis` cli to validate its config](https://support.travis-ci.com/hc/en-us/articles/115002904174-Validating-travis-yml-files).

---

Good to know.  I was worried how to debug the issue.  I had reviewed my additions to .travis.yaml, but my eye stopped at the end of the 'test' stage additions, not the line just beyond.